src/common/src/tdataformat.c

```diff
@@ -22,6 +22,26 @@
 static void tdMergeTwoDataCols(SDataCols *target, SDataCols *src1, int *iter1, int limit1, SDataCols *src2, int *iter2,
                                int limit2, int tRows, bool forceSetNull);
 
+void tdAllocMemForCol(SDataCol *pCol, int maxPoints) {
```

Comment:
Should this function return a code? This function may fail since allocation failure.